### PR TITLE
Return status of backup

### PR DESCRIPTION
**Pull request by skidelo**
I needed the return status of the pre/post backup scripts.
These changes seem to work for me, so I thought I'd push them upstream.

### CODE REVIEW

File: duplicacy-autobackup.sh [hunk 1]
```diff
@@ -24,24 +24,34 @@ do_init() {
...
-  if [[ -f $PRE_BACKUP_SCRIPT ]]; then
-    echo "Running pre-backup script"
-    sh $PRE_BACKUP_SCRIPT
-    status=$?
-  fi
-  if [[ $status != 0 ]]; then
-    echo "Pre-backup script exited with status code $status. Not performing backup." >&2
-    return
-  fi
+  while : ; do
```
Comment (christophetd) [inline duplicacy-autobackup.sh:27]: Why would you put this code inside a loop?

File: duplicacy-autobackup.sh [hunk 2]
```diff
@@ -24,24 +24,34 @@ do_init() {
...
-  duplicacy backup $DUPLICACY_BACKUP_OPTIONS
-
-  if [[ -f $POST_BACKUP_SCRIPT ]]; then
-    echo "Running post-backup script"
-    sh $POST_BACKUP_SCRIPT
+    duplicacy backup $DUPLICACY_BACKUP_OPTIONS
     status=$?
-    echo "Post-backup script exited with status $status"
-  fi
+    if [[ $status != 0 ]]; then
```
Comment (christophetd) [inline duplicacy-autobackup.sh:40]: Yup, that can make sense

**Review by christophetd — COMMENTED** [review]
Thanks for the PR. I need to fix the CI pipeline before merging anything, and I'm unsure why you put the code running pre-backup scripts inside a loop

File: duplicacy-autobackup.sh (hunk 2 above)
Comment (skidelo) [reply duplicacy-autobackup.sh:40]: It's a simple way to break out of the `do_backup` function on error.
Other options are to return early or use a bunch of nested `if` statements.

Comment (LorenzBischof) [reply duplicacy-autobackup.sh:40]: Personally I think returning early is easier to understand

**Pull request closed without merging**